src/x509_parser.rs

```diff
@@ -521,6 +551,84 @@ pub fn parse_algorithm_identifier(i: &[u8]) -> X509Result<AlgorithmIdentifier> {
     })(i)
 }
 
+fn attributes_sequence_to_map<'a>(
+    i: &'a [u8],
+    v: Vec<X509CriAttribute<'a>>,
+) -> X509Result<'a, HashMap<Oid<'a>, X509CriAttribute<'a>>> {
+    let mut attributes = HashMap::new();
+    for attr in v.into_iter() {
+        if attributes.insert(attr.oid.clone(), attr).is_some() {
+            // duplicate attributes are not allowed
+            return Err(Err::Failure(X509Error::DuplicateAttributes));
+        }
+    }
+    Ok((i, attributes))
+}
+
+fn parse_cri_attributes(i: &[u8]) -> X509Result<HashMap<Oid, X509CriAttribute>> {
+    let (i, hdr) = ber_read_element_header(i).or(Err(Err::Error(X509Error::InvalidAttributes)))?;
```

Comment:
I think using `parse_ber_tagged_implicit_g` would have been better, as it would ensure only the content of the sequence is parsed.
I will do the update